src/Rox/Core/Core.php

```diff
@@ -229,11 +231,13 @@ public function setup(
             if (!preg_match($mongoApiKeyPattern, $sdkSettings->getApiKey()) && !preg_match($uuidApiKeyPattern, $sdkSettings->getApiKey())) {
                 throw new InvalidArgumentException("Illegal rollout apikey");
             }
+
+            if (preg_match($uuidApiKeyPattern, $sdkSettings->getApiKey())) {
```

Comment:
can reuse `preg_match($uuidApiKeyPattern, $sdkSettings->getApiKey())`
put it in a parameter, use it in both lines 231 and 235